src/app/_components/Form/ControlledFormInput.tsx

```diff
@@ -3,39 +3,49 @@ import React from 'react'
 import {
   useFormContext,
   Controller,
-  RegisterOptions,
-  FieldValues,
+  type FieldValues,
+  type FieldPathByValue,
 } from 'react-hook-form'
 
 import { FormInput, type FormInputProps } from '@/components/Form/FormInput'
 
-type ControlledFormInputProps<FormType extends FieldValues> = {
-  name: keyof FormType
-  rules?: RegisterOptions
-} & FormInputProps
+type ControlledProps =
+  | 'value'
+  | 'defaultValue'
+  | 'error'
+  | 'onChange'
+  | 'onBlur'
+  | 'required'
```

Comment:
Same as for the checkbox - do we want to omit `required`? Would we nice to pass it as a prop to ControlledFormInput if needed?

Reply:
Same comment as above